.github/workflows/python_package.yml

```diff
@@ -138,6 +138,23 @@ jobs:
             method: wheel
             python_version: '3.11'
             in-ubuntu-base-container: 'true'
+          - os: LightGBM-GPU
+            container: nvcr.io/nvidia/cuda:12.9.1-devel-ubuntu22.04
+            os-display-name: ubuntu22.04-with-gpu
+            task: gpu
+            compiler: clang-17
+            method: source
```

Comment:
Intentionally only bringing over the two `method: source` GPU jobs. I think that the OpenCL build is not used as heavily now as it once was, given that we have CUDA and ROCm builds.

Having 3 jobs (2 `source` and 1 `bdist`) testing that is enough, I think. Let's use our limited CI capacity on other things.